### PR TITLE
ENH: simplify and extend rpath handling

### PR DESCRIPTION
**Pull request by dnicolodi**

### CODE REVIEW

**Comment by dnicolodi** [comment]
I'll rebase once #467 is merged

File: mesonpy/__init__.py [hunk 1]
```diff
@@ -453,31 +450,15 @@ def _install_path(
...
-                            f'@loader_path/{os.path.relpath(f".{self._project.name}.mesonpy.libs", destination.parent)}'
-                        if libdir_path not in dylib.rpath:
-                            dylib.rpath = [*dylib.rpath, libdir_path]
-                    else:
-                        # Internal libraries are currently unsupported on this platform
-                        raise NotImplementedError("Bundling libraries in wheel is not supported on platform '{}'"
-                                                  .format(platform.system()))
+            if self._is_native(os.fspath(origin)):
+                # When an executable or libray is dynamically linked to a
+                # library build as part of the project, Meson adds a library
```
Comment (rgommers) [inline mesonpy/__init__.py:455]: typo: build -> built

File: mesonpy/__init__.py [hunk 2]
```diff
@@ -453,31 +450,15 @@ def _install_path(
...
+                # load path to it poiting to the build directory, in the form of
```
Comment (rgommers) [inline mesonpy/__init__.py:456]: typo in "pointing"

File: mesonpy/__init__.py [hunk 3]
```diff
@@ -453,31 +450,15 @@ def _install_path(
...
+                # an RPATH set to @ORIGIN/ or @loader_path/. meson-python
+                # relocates the shared libraries in the $project.mesonpy.libs
```
Comment (rgommers) [inline mesonpy/__init__.py:458]: "libraries in the" -> "libraries to the"

File: mesonpy/__init__.py [hunk 4]
```diff
@@ -453,31 +450,15 @@ def _install_path(
...
+                # folder. Rewrite the the RPATH to point to that folder instead.
```
Comment (rgommers) [inline mesonpy/__init__.py:459]: typo "the the"

**Review by rgommers — APPROVED** [review]
Thanks @dnicolodi, this looks like much easier to understand code, all LGTM. Ready to merge I think. The added comment had a few typos, so please consider cleaning those up. Other than that, go ahead with merging it if you're happy too.

**Comment by dnicolodi** [comment]
Thanks for the review and for spotting the typos.

**Pull request merged**